pyproject.toml

```diff
@@ -141,7 +141,10 @@ black = "^22.6.0"
 mypy = "^0.982"
 
 setuptools = "^63.1.0"
-maturin = "~=0.14"
+
+maturin = "^0.14.10"
```

Comment:
- Version number is different from the `requires` part
- `maturin[patchelf]` only present in requires, why ?

Reply:
1. `poetry` and `python` don't resolve the same way there dependencies version

    a. for `poetry`: `^1.2.3` mean `>=1.2.3,<2.0.0`
    b. for `python`: `~=1.2.3` mean `>=1.2.3.0,==1.2.*`

    maybe I should rewrite those version requirement in `build-system` to be more clear 
2. Because, that would mean `windows` & `darwin` to install / build `patchelf` both don't support `elf` file format.
    What I would have wanted is a conditional extra (i.e.: to have the extra `patchelf` only on linux)
    Here It work because I install `patchelf` alongside `maturin` on linux and it seems to be the only dependencies listed for the extra `maturin[patchelf]`